src/input-observable.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { Component } from '@angular/core'
+import { InputObservable } from 'src/input-observable'
+import { Observable } from 'rxjs/Observable'
+import 'rxjs/add/observable/zip'
+import 'rxjs/add/operator/first'
+import 'rxjs/add/operator/skip'
+import 'rxjs/add/operator/scan'
+import 'rxjs/add/operator/take'
+import 'rxjs/add/operator/distinctUntilChanged'
+import 'rxjs/add/operator/bufferCount'
+import { Subject } from 'rxjs/Subject'
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+
+@Component({
+  selector: 'my-component',
+  template: '<h1>{{data$ | async}}</h1>',
+})
+export class MyComponent {
+  @InputObservable() public data$ = 1
+}
+
+describe('@InputObservable', () => {
+  let comp: MyComponent
+  let fixture: ComponentFixture<MyComponent>
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [MyComponent], // declare the test component
+    })
+
+    fixture = TestBed.createComponent(MyComponent)
+    comp = fixture.componentInstance // BannerComponent test instance
+  })
+
+  it('should change Input to Observable', () => {
+    expect(comp.data$ as any instanceof Observable).toBe(true)
+  })
+
+  it('should not expose internal Subject logic', () => {
+    expect(comp.data$ as any instanceof Subject).toBe(false)
```

Comment:
Does the `it should change Input to Observable` spec not forces to not expose internal Subject logic?

Reply:
`Subject<T> extends Observable<T> implements ISubscription {}`